packages/edit-navigation/src/components/name-editor/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useContext } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { MenuIdContext } from '../layout';
+import { BlockControls } from '@wordpress/block-editor';
+import {
+	__experimentalEditInPlaceControl as EditInPlaceControl,
+	ToolbarGroup,
+} from '@wordpress/components';
+
+const untitledMenu = __( '(untitled menu)' );
+export default ( { BlockEdit, blockEditProps } ) => {
+	const { saveMenu } = useDispatch( 'core' );
+	const menuId = useContext( MenuIdContext );
+	const menu = useSelect( ( select ) => select( 'core' ).getMenu( menuId ), [
+		menuId,
+	] );
+	const menuName = menu?.name ?? untitledMenu;
+	return (
+		<>
+			<BlockEdit
+				{ ...blockEditProps }
+				saveMenu={ saveMenu }
+				menuName={ menuName }
+			/>
+			<BlockControls>
+				<ToolbarGroup>
+					<EditInPlaceControl
+						label={ menuName }
+						onUpdate={ ( value ) => {
+							saveMenu( {
```

Comment:
I don't know if we should go straight to saving the menu when the name is changed. I think that's best left to the Save button. We should be able to update the name in the store. I think that'd be done using the `editEntityRecord` action from the 'core' store:

Something like:
```jsx
const { editEntityRecord } = useDispatch( 'core' );

// ...

editEntityRecord(
	'postType',
	'menu',
	menuId,
	{
		name: value || untitledMenu,
	}
);

Reply:
Considering the incoming UI updates this makes a bit more sense now. In the original designs it wasn't clear that the user should press save. In fact it's not very clear that we should save after the label is updated on the spot, most edit in place UIs update the data directly. cc @shaunandrews

Reply:
Maybe I'm thinking too much in the context of other editors where updated titles aren't saved immediately. Another thing to consider is that the toolbar input makes it feel like this is editing an attribute of the block, which wouldn't usually be considered as something that would automatically save.